lib/streamlit/credentials.py

```diff
@@ -254,3 +244,27 @@ def _exit(message):  # pragma: nocover
     """Exit program with error."""
     LOGGER.error(message)
     sys.exit(-1)
+
+
+def _get_credential_file_path():
+    return util.get_streamlit_file_path("credentials.toml")
+
+
+def _check_credential_file_exists():
+    return os.path.exists(_get_credential_file_path())
+
+
+def check_and_maybe_activate():
+    """Check credentials and potentially activate.
+
+    Note
+    ----
+    If there is no credential file and we are in headless mode, we should not
+    check, since credential would be automatically set to an empty string.
+
+    """
+    from streamlit import config
+
+    if not _check_credential_file_exists() and config.get_option("server.headless"):
+        return
+    Credentials.get_current().check_activated(auto_resolve=True)
```

Comment:
Looks like `check_activated` is only ever called here, so:
* prefix it with `_` (underscore)
* Remove the `auto_resolve` parameter, since the only place that calls this function has it set to `True`.

Reply:
Tests calls this function with the default which is `auto_resolve=False`. Likely not a best practice though...

Reply:
I am changing the default around.